### DISCUSSION TITLE
Possibility to specify the test suites for pomless tests?

### DISCUSSION THREAD

**Discussion by vogella** [eclipse-tycho/tycho#3105] [Q&A]
AFAIK for test bundles specifying the test classes or suites I have to define pom. Is this still correct?

**Comment by laeubi** [comment]
@vogella for such purpose you usually have a plugin (management) configuration in your configurator pom (usually at project root), where you define a property for that what you want to configure in the pomless one, maybe with a default.

Lets say we want to configure this here:
https://tycho.eclipseprojects.io/doc/latest/tycho-surefire-plugin/test-mojo.html#testClass

so in the configurator pom have something like that:

```
<project>
  ...
  <build>
    <pluginManagement>
      <plugins>
        <plugin>
          <groupId>org.eclipse.tycho</groupId>
          <artifactId>tycho-surefire-plugin</artifactId>
          <configuration>
             <testClass>${surefireTestSuite}</testClass>
          </configuration>
        </plugin>
        ...
      </plugins>
    </pluginManagement>
```

Now you can leverage property overrides as described here: https://github.com/eclipse-tycho/tycho/wiki/Tycho-Pomless#define-properties and put this into your `build.properties`

```
pom.model.property.surefireTestSuite = my.pomless.TestSuite
```

**Comment by vogella** [comment]
That is cool, thanks @laeubi for the explanation.  Once my JDT pomless contributions are in and the aggregator build is stable again, I plan to apply this learning to JDT to further simplify their build setup.
